pkg/apis/core/validation/validation.go

```diff
@@ -4353,6 +4353,12 @@ func ValidateConfigMap(cfg *core.ConfigMap) field.ErrorList {
 		}
 		totalSize += len(value)
 	}
+	for key, value := range cfg.BinaryData {
```

Comment:
also need to ensure there is no key duplication between data and binarydata maps

Reply:
Please see note from @smarterclayton here, there was an ask to remove that validation i believe:
https://github.com/kubernetes/kubernetes/pull/33549/commits/9ba0da82bba6596766672b2146be01abc576277b#r100961600

Reply:
hmm... having the same key in both `data` and `binaryData` means the object is undefined...

also, from https://github.com/kubernetes/kubernetes/pull/33549#discussion_r100961540:
> Needs a much more descriptive comment, including a description of the validation rules (must not overlap with data)

Reply:
@smarterclayton can you please chime in?